source/Octopus.Tentacle/Scripts/ScriptStateStore.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using Newtonsoft.Json;
+using Octopus.Tentacle.Contracts;
+using Octopus.Tentacle.Util;
+
+namespace Octopus.Tentacle.Scripts
+{
+    public class ScriptStateStore : IScriptStateStore
+    {
+        readonly SemaphoreSlim scriptStateLock = new(1, 1);
+        readonly IScriptWorkspace workspace;
+        readonly ScriptTicket scriptTicket;
+        readonly IOctopusFileSystem fileSystem;
+
+        public ScriptStateStore(ScriptTicket scriptTicket, IScriptWorkspace workspace, IOctopusFileSystem fileSystem)
+        {
+            this.scriptTicket = scriptTicket;
+            this.workspace = workspace;
+            this.fileSystem = fileSystem;
+        }
+
+        string StateFilePath => workspace.ResolvePath("scriptstate.json");
+
+        public ScriptState Create()
+        {
+            using var _ = scriptStateLock.Lock();
```

Comment:
Just curious. We are using synchronous primitives here. Would `lock(object)`, which doesn't require a customer releaser, work here? Are you preparing to make Tentacle Async?

Reply:
It wasn't intentionally done to support any future changes.